packages/angular_devkit/core/src/jobs/README.md

```diff
@@ -0,0 +1,386 @@
+1. `JobState.Created`. The job's dependencies (see "Synchronizing and Dependencies") are done 
+    running, and the job's argument will be validated before the Job's code will be executed.
+1. `JobState.Started`. The job has been started. The job's code is then called.
+1. `JobState.Ended`. The job has ended and is done running.
+
+Each state (except Queued) corresponds to a JobEvent on the `outputChannel` observable that 
+triggers the state change. The `Queued` state is set as the job is scheduled, so there is no need
+to listen to an event. A user can listen to these events, or use the `state` member which will
+correspond to one of those. The Job implementation itself should not emit those events (they 
+will be filtered out); the Scheduler should take care of it.
+
+## `Job<OutputType>` Object
+The `Job` object is returned when scheduling a job. It contains multiple members and utilities to 
+ease the usage of jobs.
+
+1. `id`. A unique symbol that can be used as a Map key.
+1. `description`. The description of the job from the scheduler. See `JobDescription` object.
+1. `argument`. The argument value that was used to start the job.
+1. `input`. An `Observer` that can be used to send validated inputs to the Job itself.
+1. `output`. An `Observable<OutputType>` that filters out events to get only the returned output
+    of a job.
+1. `promise`. A promise that waits for the last output of a job. Returns the last value outputted
+    (or no value if there's no last value).
+1. `state`. The current state of the job (see `LifeCycle`).
+1. `progress`. A number between 0 and 1 that can be reported by the job as the progress of the job.
+1. `channels`. A map of side channels the user can listen to as `Observable`.
+1. `ping()`. A function that can be used to ping the job, receiving a `Promise` for when the ping
+    is answered.
+1. `stop()`. Sends a `stop` input to the job, which suggests to stop the job. The job itself can
+    choose to ignore this event.
+1. `inputChannel`. The raw input channel.
+1. `outputChannel`. The raw output channel.
+
+# Creating Jobs
+
+A job is at its core a function with a Schema associated to it;
+
+```typescript
+import { Observable } from 'rxjs';
+import { jobs } from '@angular-devkit/core';
+
+const argument = {
+  type: 'array', items: { type: 'number' },
+};
+const output = {
+  type: 'number',
+};
+
+export function add(argument: number[]): Observable<jobs.JobEvent<number>> {
+  return new Observable(o => {
+    o.next({
+      kind: jobs.JobEventKind.Output,
+      output: argument.reduce((total, curr) => total + curr, 0),
+    });
+    o.complete();
+  });
+}
+
+// Add a property to `add` to make it officially a JobHandler. The Job system does not recognize any
+// function as a JobHandler.
+add.jobDescription = {
+  argument: argument,
+  output: output,
+};
+
+// Call the job with an array as argument, and log its output.
+declare const scheduler: jobs.Scheduler;
```

Comment:
`declare` modifier is incorrect here.

Reply:
Why? It's a declaration. This is an example of usage, not actual code.